src/olympia/versions/models.py

```diff
@@ -796,10 +796,15 @@ def disable_old_files(self):
                 f.update(status=amo.STATUS_DISABLED)
 
     def reset_due_date(self, due_date=None):
-        if not self.due_date or due_date:
-            due_date = due_date or get_review_due_date()
-            # We need signal=False not to call update_status (which calls us).
-            self.update(due_date=due_date, _signal=False)
+        if self.should_have_due_date:
```

Comment:
Something to keep in mind - which maybe could help tests if we need an override mechanism there - is that we're going to need something to "keep" the `due_date` if it was set explicitly by an admin in https://github.com/mozilla/addons-server/issues/20152

Reply:
I was thinking there would be a "human review requested" flag/date that could be added or reused.

Reply:
Yeah that makes sense.